backend/downsample.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2020 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+"""Downsample strategies.
+
+Contains all of the downsample strategies. Use downsample(), and
+secondary_downsample() for downsampling records stored in files.
+"""
+
+import utils
+
+
+SECOND_TO_MICROSECOND = 1E6
+FLOAT_PRECISION = 4
+
+
+def triangle_area(point1, point2, point3):
```

Comment:
naming convention: 
https://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations

This is a internal method, should name it like _triangle_area()

Please also fix the name of other methods in the file.

Reply:
I assume that only methods being called directly outside of package is defined as public method. Does it mean that I should maybe rename all methods in downsample.py to start with `_`, except for `downsample`?

Reply:
yes.